src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,793 @@
+    this.#borderSizeRAF.perform(this.#updateBorderSize);
+  };
+
+  #updateBorderSize = () => {
+    // - - - - - - - - - - - - - DOM read operations - - - - - - - - - - - - -
+    const computedStyle = getComputedStyle(this.el);
+
+    // - - - - - - - - - - - - - DOM write operations - - - - - - - - - - - - -
+    this.el.style.setProperty(
+      DIALOG_BORDER_BLOCK_START_SIZE,
+      computedStyle.borderBlockStartWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_BLOCK_END_SIZE,
+      computedStyle.borderBlockEndWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_INLINE_START_SIZE,
+      computedStyle.borderInlineStartWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_INLINE_END_SIZE,
+      computedStyle.borderInlineEndWidth
+    );
+  };
+
+  #removeBorderSizeWatcher = () => {
+    if (this.#borderSizeObserver) {
+      this.#borderSizeObserver.disconnect();
+      this.#borderSizeObserver = null; // Free the memory
+    }
+
+    this.#borderSizeRAF = null; // Free the memory
+  };
+
+  #fixDialogPosition = () => {
+    const dialogRect = this.#dialogRef.getBoundingClientRect();
+    const inlineStart = this.#isRTLDirection
+      ? document.documentElement.getBoundingClientRect().width -
+        (dialogRect.left + dialogRect.width)
+      : dialogRect.left;
+    const blockStart = dialogRect.top;
+
+    setProperty(
+      this.#dialogRef,
+      DIALOG_INLINE_START,
+      inlineStart - this.#draggedDistanceX
+    );
+    setProperty(
+      this.#dialogRef,
+      DIALOG_BLOCK_START,
+      blockStart - this.#draggedDistanceY
+    );
+  };
+
+  #unfixDialogPosition = () => {
+    this.#dialogRef.style.removeProperty(DIALOG_INLINE_START);
+    this.#dialogRef.style.removeProperty(DIALOG_BLOCK_START);
+  };
+
+  render() {
+    const dialogParts = this.showHeader ? "dialog has-header" : "dialog";
+
+    return (
+      <Host
+        class={{
+          "gx-dialog-header-drag": !this.hidden && this.allowDrag === "header",
+          [RESIZING_CLASS]: this.resizing
+        }}
+      >
+        <dialog
+          part={dialogParts}
+          onClose={this.#handleDialogToggle}
+          ref={el => (this.#dialogRef = el)}
+          onMouseDown={this.allowDrag === "box" ? this.#handleMouseDown : null}
+        >
+          {this.showHeader && (
+            <header
+              class="header"
+              part="header"
+              onMouseDown={
+                this.allowDrag === "header" ? this.#handleMouseDown : null
+              }
+            >
+              <slot name="header-start" />
+              {this.caption && (
+                <h2 part="caption" class="caption">
+                  {this.caption}
+                </h2>
+              )}
+              <slot name="header-end" />
+              <button
+                part="close"
+                class="close-button"
+                onClick={this.#closeHandler}
+              ></button>
+            </header>
```

Comment:
Some changes to improve accessibility and code readability. For reference:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role


```suggestion
        <dialog
          aria-labelledby={this.caption ? "heading" : null}
          part={dialogParts}
          onClose={this.#handleDialogToggle}
          onMouseDown={this.allowDrag === "box" ? this.#handleMouseDown : null}
          ref={el => (this.#dialogRef = el)}
        >
          {this.showHeader && (
            <div
              class="header"
              part="header"
              onMouseDown={
                this.allowDrag === "header" ? this.#handleMouseDown : null
              }
            >
              {this.caption && (
                <h2 id="heading" part="caption" class="caption">
                  {this.caption}
                </h2>
              )}

              <button
                class="close-button"
                part="close-button"
                type="button"
                onClick={this.#closeHandler}
              ></button>
            </div>
```

Reply:
Thanks 👍